### ISSUE TITLE
Allow commenting on lines not changed in a PR

### ISSUE THREAD

**Issue by stuartleeks** [microsoft/vscode-pull-request-github#3305]
I often find myself wanting to comment on lines that haven't changed in a PR. This is typically to call out other places that need updating based on the changes in the PR, but that had been missed.

From what I can tell, the existing behaviour only allows comments to be added on lines that were changed in the PR?

**Comment by albertodiazdorado** [comment]
We would also be very interested in this feature. The limitation forces us into very bad patterns, such as leaving a general comment referencing file and line (as if we did not have a UI for reviews)

**Comment by alexr00** [comment]
We do have a way to make a file level comment, but GitHub still doesn't allow us to leave a comment on a specific line that's outside of a PR:

![Image](https://github.com/user-attachments/assets/c9e80f7b-356a-4c37-87ef-b79b9a4c8c2b)

**Comment by rmckewen** [comment]
I and most everyone I know would also like to see this feature made available

**Comment by eyal-converge** [comment]
Hi @alexr00 any updates ?

**Comment by yashwant-raut** [comment]
FWIW, here’s a simple hack I follow:

I select multiple lines in the file (even if they are not part of the current PR). When you select lines, the URL in the browser updates to include the selected line numbers. If you copy this URL and open it in another tab, you'll see the selected lines highlighted.  
This way, when you share the link with someone (for example, while reviewing a PR), they will see the same lines highlighted.

**Comment by Malix-Labs** [comment]
GitHub made it available recently!

https://github.blog/changelog/2025-09-25-pull-request-files-changed-public-preview-now-supports-commenting-on-unchanged-lines/

---

_By the way, alex was right about that part, i don't know why some people were booing her, it was just a fact:_

[<img width="1060" height="257" alt="Image" src="https://github.com/user-attachments/assets/ff407fcb-0065-4371-bd2d-4407d7eb8534" />](https://github.com/microsoft/vscode-pull-request-github/issues/3305#issuecomment-1035034122)

**Comment by alexr00** [comment]
Yes, we are looking into it! GitHub hasn't yet made API available for this, but they're working on it. Soon ™️.

**Comment by hilmi-yilmaz** [comment]
Hi, just checking in on this feature request. Is there any update of ETA we can expect? Thanks in advance!

**Comment by Malix-Labs** [comment]
^ or potentially a way we can help the team shipping this feature?

**Comment by alexr00** [comment]
As far as I understand, the GitHub folks have most of the work in place, they just need to finalize the API. I've pinged them (again). (This is the 5th most upvoted issue `microsoft/vscode-pull-request-github` at this point. As soon as I have to ability to implement this I plan to do so).

**Comment by imanmahjoubi** [comment]
This is completed ☑️

**Comment by alexr00** [comment]
👋I've gotten to try out the API and provide feedback (https://github.com/microsoft/vscode-pull-request-github/pull/8573). The API is still not rolled out to everyone so I can't merge it yet.

**Comment by harisonw** [comment]
Hopefully GitHub pull through with the support for this

**Comment by BosTheCoder** [comment]
Hey @alexr00  is there any update on this by any chance?

**Comment by alexr00** [comment]
Still waiting on GitHub to release the API changes 😢
